tools/codegen/api/types.py

```diff
@@ -266,11 +266,16 @@ def no_default(self) -> 'Binding':
             argument=self.argument,
         )
 
-    def decl(self) -> str:
+    def decl(self, *, func_ptr_cast: bool = False) -> str:
```

Comment:
So... what does `func_ptr_cast` do anyway?

Reply:
Why isn't this just not a separate method on its own? (Or even just self.type?)

Reply:
yeah it totally should be

Reply:
oh, `DispatcherSignature` already has a `type` function that does everything I need except making the type a pointer. I'll add a flag there.

E.g.
`at::Tensor(const at::Tensor &, const at::Tensor &, const at::Scalar &)`
vs
`at::Tensor (*)(const at::Tensor &, const at::Tensor &, const at::Scalar &)`

Reply:
oh yeah, that's kind of unfortunate. I don't think there's any way to do it compositionally without breaking out using

```
using f = int(int);
using fptr = f*;
```

but that's inconvenient for codegen, so this will have to do.